sqlmesh/core/engine_adapter/base.py

```diff
@@ -2040,6 +2040,12 @@ def temp_table(
         Yields:
             The table expression
         """
+        if isinstance(name, str):
+            name = exp.to_table(name)
+        # ensure that we use default catalog if none is not specified
+        if isinstance(name, exp.Table) and not name.catalog:
```

Comment:
you may want to check db as well, if table.db, otherwise you'll add a catalog if there's no db...

Reply:
Ok I'm adding it but I would have expected that it would be projected only if both were defined. Isn't it checked at "print" level?